### PR TITLE
enhance: combobox with nested groups & use for default alias dropdown

### PR DESCRIPTION
**Pull request by ivyjeong13**
* get ComboBox working with nested groups and filtering with nested groups/items
* change default model alias selects to comboboxes

Screenshot 2024-12-20 at 2 23 03 PM

Screenshot 2024-12-20 at 2 26 25 PM

### CODE REVIEW

File: ui/admin/app/components/composed/ComboBox.tsx [hunk 1]
```diff
@@ -25,22 +25,25 @@ type BaseOption = {
...
 
 type ComboBoxProps<T extends BaseOption> = {
     allowClear?: boolean;
     clearLabel?: ReactNode;
+    emptyLabel?: ReactNode;
     onChange: (option: T | null) => void;
-    options: T[] | GroupedOption<T>[];
+    options: (T | GroupedOption<T>)[];
     placeholder?: string;
+    suggested?: string[];
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/composed/ComboBox.tsx]: `suggested` feels like an implementation detail that should be handled by the parent. Could we instead expose a `renderItem` prop that allows the parent to implement each row as it pleases?


```suggestion
    renderItem?: (item: T) => ReactNode;
```

File: ui/admin/app/components/composed/ComboBox.tsx [hunk 2]
```diff
@@ -94,16 +112,85 @@ function ComboBoxList<T extends BaseOption>({
...
+
+        // If both or neither are suggested, maintain original order
+        if (aIsSuggested === bIsSuggested) return 0;
+        // Sort suggested items first
+        return aIsSuggested ? -1 : 1;
+    };
+
+    const handleValueChange = (value: string) => {
+        setFilteredOptions(filterOptions(options, value));
+    };
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/composed/ComboBox.tsx]: same here. this should be handled by the parent. Especially if we ever need a combobox that displays data paginated by the server

File: ui/admin/app/components/composed/ComboBox.tsx [hunk 3]
```diff
@@ -94,16 +112,85 @@ function ComboBoxList<T extends BaseOption>({
...
+                        .includes(searchValue.toLowerCase());
+                const isGroupHeadingMatch =
+                    "heading" in option &&
+                    option.heading
+                        .toLowerCase()
+                        .includes(searchValue.toLowerCase());
+
+                if (isGroupHeadingMatch || isSingleValueMatch) {
+                    return [...acc, option];
+                }
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/composed/ComboBox.tsx:143]: Absolutely no change necessary here! But a fun alternative...

```ts
                const hasMatch = [
                    "name" in option && option.name,
                    "heading" in option && option.heading,
                ]
                    .filter((x) => !!x)
                    .join("|")
                    .toLowerCase()
                    .includes(searchValue.toLowerCase());

                if (hasMatch) {
                    return [...acc, option];
                }

```

File: ui/admin/app/components/composed/ComboBox.tsx (hunk 2 above)
Comment (ivyjeong13) [reply ui/admin/app/components/composed/ComboBox.tsx]: 🙌 Good call out, moved the suggestions out of combobox! Thanks!

**Review by ryanhopperlowe — COMMENTED** [review]
looking good

**Review by ryanhopperlowe — APPROVED** [review]

**Pull request merged**